app/api/routers/position.py

```diff
@@ -285,21 +319,24 @@ def __call__(
 
         resp_data = []
         for lock_event in lock_events:
-            resp_data.append(
-                {
-                    "category": lock_event[0],
-                    "transaction_hash": lock_event[1],
-                    "token_address": lock_event[2],
-                    "lock_address": lock_event[3],
-                    "account_address": lock_event[4],
-                    "recipient_address": lock_event[5],
-                    "value": lock_event[6],
-                    "data": lock_event[7],
-                    "block_timestamp": lock_event[8]
-                    .replace(tzinfo=UTC)
-                    .astimezone(local_tz),
-                }
-            )
+            event_data = {
+                "category": lock_event[0],
+                "transaction_hash": lock_event[1],
+                "token_address": lock_event[2],
+                "lock_address": lock_event[3],
+                "account_address": lock_event[4],
+                "recipient_address": lock_event[5],
+                "value": lock_event[6],
+                "data": lock_event[7],
+                "block_timestamp": lock_event[8]
+                .replace(tzinfo=UTC)
+                .astimezone(local_tz),
+            }
+            if request_query.include_token_details is True:
+                event_data["token"] = self.token_model.from_model(
+                    lock_event[9]
+                ).__dict__
```

Comment:
- If **include_token_details=true** is queried, token_detail attribute is given.